2018-komp-ling/projects/serikov.md

```diff
@@ -0,0 +1,75 @@
+# Compling project proposal
+
+## Abstract
+RE: extracting phonological rules from neural networks paper: Jennifer Rodd (1997) "Recurrent Neural-Network Learning of Phonological Regularities in Turkish". CoNLL97: Computational Natural Language Learning, http://www.aclweb.org/anthology/W97-1012
+
+## Introduction
+**TO-DO** the background of the problem to be solved or topic to be studied, with references for the low-background reader.
+
+## Proposed goals
+
+### MVP
+* Re-implement the paper
+* Release the code and documentation showing how to run it
+* Extract the graphs that are contained in the original paper
+
+### EP
+Run the NNs explored in the original paper with other languages in the Turkic family to see if their networks split the same way (other Turkic languages have different vowel harmony systems, and Uzbek does not have vowel harmony at all). 
+
+### HAP
+Apply the received knowledge of computational phonology and Turkic family languages phonology in particular to solve the task of phonetic embeddings alignment without having big parallel corpus.
+
+## Requirements
+
+**NB!** Each sub-goal is accompanied with the documentation composing and sources publishing where needed.
+
+**NB!** Goals may be changed on the run if necessary (e.g. smth unexpected discovered). All the possible changes should be discussed with mentors.
+
+### MVP requirements
+#### Skills required
+* Brief knowledge of actual NN creating techniques.
+* Data preprocessing skills
+* Vizualization skill.
+
+#### Sub-goals
+* 1 week| Reproduce the dataset used in original paper
```

Comment:
"The input to the network is a series of sequentially presented phonemes from a corpus of 602  Turkish  words. "

This shouldn't take any time at all. I can provide you with the words.

Reply:
this week the input data reproduction took ~3 days, and there still being some questions unanswered, so I think that weekly buffer to deal with the possible problems with the data collection could be helpful.